extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -222,15 +223,15 @@ void registerAutoSecurityFilter(BuildProducer<SyntheticBeanBuildItem> syntheticB
             SmallRyeOpenApiConfig openApiConfig,
             OpenApiFilteredIndexViewBuildItem apiFilteredIndexViewBuildItem,
             List<SecurityInformationBuildItem> securityInformationBuildItems,
-            OpenApiRecorder recorder) {
+            OpenApiRecorder recorder,
+            HttpBuildTimeConfig httpConfig) {
+
         OASFilter autoSecurityFilter = null;
-        if (openApiConfig.autoAddSecurity) {
+
+        if (openApiConfig.autoAddSecurity
```

Comment:
Now you are able to enable security with `io.quarkus.smallrye.openapi.common.deployment.SmallRyeOpenApiConfig#autoAddSecurity` and including security extension and either use RolessAlowed-Authenticated or HTTP permissions to have security enabled, correct?

I wonder what happens when user uses other HTTP Security Policies like Keycloak Authorization or has their own one because they implemented global policy due to our limited path matching capabilities? Users will have to specify `securityScheme` when they want it generated, rather then it being detected?

I can see why this is an improvement, but it will blocking moving HTTP permissions to runtime. Isn't that rather rare to have security extension added (you need to add it explicitly, right?) without securing anything? Why can't you simply add it always when security extension is present and if user don't want to add these schemes, they will set `auto-add-security` to false.

Also, the config property description says nothing about actually securing endpoints, therefore my proposed change would conform description `This will automatically add security based on the security extension included (if any).`. If you don't agree, please change the description.

Reply:
> Now you are able to enable security with `io.quarkus.smallrye.openapi.common.deployment.SmallRyeOpenApiConfig#autoAddSecurity` and including security extension and either use RolessAlowed-Authenticated or HTTP permissions to have security enabled, correct?

@michalvavrik , that's correct. For your question on other policies, I think it depends on what `HttpBuildTimeConfig` and the `SecurityInformationBuildItem` look like for other/custom security policies. I'm thinking that the answer is what you said - the user would need to specify their own `securityScheme` and this approach only works when `SecurityInformationBuildItem` has one of the known `SecurityModel`s.

Reply:
Thanks for reply @MikeEdgar. I don't see any good reason why scheme shouldn't be generated when the security extension is present and `SecurityInformationBuildItem` is present as well. Just because someone doesn't use HTTP permissions and RBAC annotations, it doesn't mean they don't do security. Is there a point why you want to check HTTP permissions explicitly?

Reply:
> Is there a point why you want to check HTTP permissions explicitly?

Only because it's what I needed for another project, but I think some more general approach would be better if it's possible.

Reply:
Well, it's not a deal break, I suppose we can still migrate HTTP permissions and just add some flag for OpenAPI, but I'd appreciate if it was possible to not use HTTP permissions during the build time unless you need to. Whatever works for you, thanks.

Reply:
> Why can't you simply add it always when security extension is present and if user don't want to add these schemes, they will set `auto-add-security` to false.

This seems like a reasonable approach. @phillip-kruger wdyt? This would remove the check for `@RolesAllowed` and `@Authenticated` when producing the auto-security filter.

Reply:
The auto security filter will run at build-time in the absence of the annotations currently, so that may be another argument to do the same for the run-time filter. I'll update the PR with that change for discussion.

Reply:
I am not following everything, but we should not add security automatically if none is used, even if there is security module added. If we could not detect this automatically, then the user can still configure or annotate. The auto feature is an OpenAPI feature, that tries to reduce the OpenAPI annotation/configuration needed if we can safely detect security. But adding it if none is detected is annoying and was a bug #34700  I fixed here #36279

Reply:
> I am not following everything, but we should not add security automatically if none is used, even if there is security module added. If we could not detect this automatically, then the user can still configure or annotate. The auto feature is an OpenAPI feature, that tries to reduce the OpenAPI annotation/configuration needed if we can safely detect security. But adding it if none is detected is annoying and was a bug #34700 I fixed here #36279

I agree with the proposed solution, but I'd like to point out that question _I didn't setup anything related to security - so why it's there?_ in the #34700 has answer: configuration property https://quarkus.io/guides/all-config#quarkus-smallrye-openapi_quarkus.smallrye-openapi.auto-add-security says literally _This will automatically add security based on the security extension included (if any)._. So Clement had there security scheme because he included the Security extension. I tried to point out that HTTP permissions and RBAC annotations are not only ways to secure endpoints (e.g. KC authorizer), HTTP permissions are one of the policies.

Reply:
Maybe we should fix the configuration property description then :)

Reply:
No objections there :-) I simply relied on docs in this for that's as much as I know about this thing.

Reply:
Auto security adds security (in OpenAPI) when nothing is configured in OpenAPI (annotations, configuration) but only if
- Security extensions are added
- Other (security) annotations and configuration are detected.

The second part are not in the description ....

Reply:
Ok, got it :+1:. I think @michalvavrik's suggestion earlier was to remove the second part from the extension's code and allow users not wanting security to disable it using `false`. The issue I see is that if we require annotations, there are other possible ways to setup security that aren't supported.

Reply:
Also remember, auto security for OpenAPI only run if there is not OpenAPI Security annotations. The user can always still add their own OpenAPI Security annotation when it's a special case.